### ISSUE TITLE
Access log entries are not logged in istio sidecar for ingress traffic

### ISSUE THREAD

**Issue by aladdin-atypon** [istio/istio#39734]
### Bug Description

I have alb ingress which routes its traffic to istio-ingressgateway.

From there I have a gateway:

    ---
    apiVersion: networking.istio.io/v1alpha3
    kind: Gateway
    metadata:
      name: "X-gateway"
      namespace: dev
    spec: 
      selector:
        istio: ingressgateway
      servers:
      - port:
          number: 80
          name: http
          protocol: HTTP
        hosts:
        - "dev.xxx.com"

Also I have the virtual service in place:

    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: vs-istio-ingress
      namespace: dev
    spec:
      gateways:
        - X-gateway
      hosts:
        - "dev.xxx.com"
      http:
         - route:
            - destination:
                host: serviceX
                port:
                  number: 8080

From there I have the service defined:

    apiVersion: v1
    kind: Service
    metadata:
      namespace: dev
      name: serviceX
      labels:
        app: appX
    spec:
      selector:
        app: podX
      ports:
        - port: 8080

I have access log enabled in the operator by setting:

    spec:
      meshConfig:
        accessLogFile: /dev/stdout

The issue is when I hit the service from the ingress, the ingressgateway itself has the access log entry there with response code 200 and I can see the access log entry in the app access log as well, but not the sidecar access log! (it's single pod), however, when the request is coming to the service via one of the service mesh the log entry is there in sidecar proxy access log and app access log as well.

### Version

```prose
istioctl version


client version: 1.10.0
control plane version: 1.10.0
data plane version: 1.10.0 (271 proxies)

kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.1
Kustomize Version: v4.5.4
Server Version: v1.21.4


helm version --short
v3.9.0+g7ceeda6
```

### Affected product area

- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [X] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by zirain** [comment]
can your share the accesslog from ingressgateway?

**Comment by aladdin-atypon** [comment]
Thanks @zirain ! updated.

**Comment by aladdin-atypon** [comment]
The issue was here:

spec:
  selector:
    app: podX
  ports:
    - port: 8080
    
The port should be named.

Anyway, @zirain , can you please point out to the source code of the access log logic? I've been a developer for long time and I believe it's better to debug than post questions around :)

**Comment by zirain** [comment]
https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/

**Comment by aladdin-atypon** [comment]
I asked for a clue about the source code of access log not anything related to a documentation since the issue is already addressed and obviously using the above doc. is something I read to address the issue already.

**Comment by zirain** [comment]
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/accesslog.go

**Comment by zirain** [comment]
from the ingressgateway log you provide, there's nothing about upstream, do you custom the JSON output?

**Comment by aladdin-atypon** [comment]
Nope, I haven't, as I mentioned above, the issue was solved by naming the service port and as you provided in the doc.
Thank you @zirain !
